src/chart/components/y_axis/y-axis-global.component.ts

```diff
@@ -0,0 +1,106 @@
+import { CanvasBoundsContainer } from '../../canvas/canvas-bounds-container';
+import { BarType, FullChartColors, FullChartConfig } from '../../chart.config';
+import { ClearCanvasDrawer } from '../../drawers/clear-canvas.drawer';
+import { CompositeDrawer } from '../../drawers/composite.drawer';
+import { DrawingManager } from '../../drawers/drawing-manager';
+import { PriceMovement } from '../../model/candle-series.model';
+import { CanvasModel } from '../../model/canvas.model';
+import { ChartBaseElement } from '../../model/chart-base-element';
+import { DataSeriesType } from '../../model/data-series.config';
+import { PaneManager } from '../pane/pane-manager.component';
+import {
+	resolveColorForArea,
+	resolveColorForBar,
+	resolveColorForBaseLine,
+	resolveColorForCandle,
+	resolveColorForHistogram,
+	resolveColorForLine,
+	resolveColorForScatterPlot,
+	resolveColorForTrendAndHollow,
+	resolveDefaultColorForLabel,
+} from './label-color.functions';
+import { YAxisPriceLabelsDrawer } from './price_labels/y-axis-price-labels.drawer';
+import { YAxisDrawer } from './y-axis.drawer';
+
+export type LabelColorResolver = (priceMovement: PriceMovement, colors: FullChartColors) => string;
+
+/**
+ * Contains common y-axis stuff, like drawers & label color resolver
+ */
+export class YAxisGlobalComponent extends ChartBaseElement {
```

Comment:
I'm not sure about the name for this one, but there should be some entity like this

Reply:
**YAxisCommon, YAxisUtilities**. But do we really need so global component? May be we can separate it's functionalily?

Reply:
We create drawers here and there's also label color resolver, it seems like that this kind of staff is common , and there is no need to create a separate for each y-axis

Reply:
maybe it could, I just thought that there is no need to have separate `labelColorResolvers` for each extent